internal/agent/agent_test.go

```diff
@@ -702,3 +711,107 @@ func BenchmarkStorePluginOutput(b *testing.B) {
 		})
 	}
 }
+
+func Test_ProcessSampling_FeatureFlagIsEnabled(t *testing.T) {
+	cnf := &config.Config{
+		IncludeMetricsMatchers: map[string][]string{"process.name": {"some-process"}},
+	}
+	someSample := struct {
+		evenType string
+	}{
+		evenType: "ProcessSample",
+	}
+	a, _ := NewAgent(cnf, "test", test.NewFFRetrieverReturning(true, true))
+
+	// when
+	actual := a.Context.shouldIncludeEvent(someSample)
+
+	// then
+	assert.Equal(t, true, actual)
+}
+
+// ProcessSample is mimicking the real ProcessSample which we can't reference from here due to import cycles.
```

Comment:
It is possible to fix the import cycles instead of mimicking?

Reply:
of course! this is crap

Reply:
This is awesome!! Thanks!